### PR TITLE
Mailgun and create an event

### PR DESCRIPTION
**Pull request by Liudmila02**

### CODE REVIEW

File: api/src/controller/event.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  }
+}
+
+export const UpdateEvent = async (req, res) => {
+  const { description, date, time, place } = req.body
+  try {
+    const event = await db.events.findOne({ where: { id: req.params.eventId } })
+    console.log(req.params.eventId)
+    if (!event) throw new Error()
+    const updatedEvent = event.update({
```
Comment (Djnt) [inline api/src/controller/event.js]: columns here updating even if there are no changes. try assert key-value pair with spread if it's present,
like ...(description && {description}),   . . .

File: api/src/controller/event.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    return res.status(400).send({
+      message: 'Event Not Update',
+    });
+  }
+}
+
+export const ShowEvent = async (req, res) => {
+  try {
+    const newEvent = await db.events.findOne({ where: { id: req.params.eventId } })
+    console.log(req.params.eventId)
```
Comment (Djnt) [inline api/src/controller/event.js]: remove

File: api/src/controller/event.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    res.status(200).json({
+      event: newEvent
+    })
+    console.log(newEvent)
+  } catch (err) {
+    console.log(err)
```
Comment (Djnt) [inline api/src/controller/event.js]: remove all logs

File: api/src/controller/event.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    })
+  } catch (err) {
+    console.log(err)
+    return res.status(400).send({
+      message: 'Event Not Update',
+    });
+  }
+}
+
+export const ShowEvent = async (req, res) => {
```
Comment (Djnt) [inline api/src/controller/event.js]: would be better to pass only eventId here, and response is also bad here, you should just pass result to route, and route should handle it as response.

File: api/src/db/migrations/20200103115544-event.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    await queryInterface.sequelize.query(
```
Comment (Djnt) [inline api/src/db/migrations/20200103115544-event.js]: you should include this only in first migration

File: api/src/utils/validator.js [hunk 6]
```diff
@@ -21,3 +21,26 @@ export const validateRegisterForm = args => {
...
+export const validateEventForm = args => {
+  let Errors = {}
+  if (!args.description) Errors.description =  "this field cann't be blank"
+  if (!args.date) Errors.date =  "choose a date"
+  if (!args.time) Errors.time =  "choose time"
+  if (!args.place) Errors.place =  "choose a place"
+  
+  if (!args.description || args.description.length < 4)
+    Errors.description =
+      "password can't be blank or shorter than 8 chars"
```
Comment (Djnt) [inline api/src/utils/validator.js]: looks wrong

File: api/src/controller/event.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+      ...(date && { date }),
+      ...(time && { time }),
+      ...(place && { place })
+    })
+    res.status(200).send({
+      message: 'Event successfully update',
+      event: updatedEvent
+    })
+  } catch (err) {
+    console.log(err)
```
Comment (eugenekorpan) [inline api/src/controller/event.js]: do we need this here?

File: api/src/passport/user.js [hunk 8]
```diff
@@ -29,30 +29,32 @@ export const registerUser = async args => {
...
+    const senderPass = process.env.SENDER_EMAIL_PASSWORD 
+
     client.set(token, args.email, redis.print)
     client.expire(token, hoursFromNow)
-
-    var nodemailer = require('nodemailer')
-
-    var transporter = nodemailer.createTransport({
-      service: 'gmail',
+    console.log(token)
```
Comment (eugenekorpan) [inline api/src/passport/user.js]: yeah, please remove all console.log() before pushing to github

File: api/src/passport/user.js [hunk 9]
```diff
@@ -29,30 +29,32 @@ export const registerUser = async args => {
...
+    const nodemailer = require('nodemailer')
+    const transporter = nodemailer.createTransport({
+      host: "smtp.mailgun.org",
```
Comment (eugenekorpan) [inline api/src/passport/user.js]: I guess configs like these should be in a separate, config file, no?

File: api/src/passport/user.js [hunk 10]
```diff
@@ -103,3 +105,6 @@ export function hashPassword(password) {
   const salt = bcrypt.genSaltSync(10)
   return bcrypt.hashSync(password, salt)
 }
+
+
```
Comment (eugenekorpan) [inline api/src/passport/user.js]: unnecessary blank lines at the end of the file

File: api/src/router/event.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+import { ShowEvent } from '../controller/event'
+
+import { isAuthenticated }  from '../router/profile'
+const router = new Router()
+
+router.post('/api/event', isAuthenticated, async (req, res)=>{
+  let result = await newEvent(req.body, req)
+  if (result.success == true) {
+    res.status(200).json(result.event)
+  }else {
```
Comment (eugenekorpan) [inline api/src/router/event.js]: space missing

File: api/src/utils/validator.js [hunk 12]
```diff
@@ -21,3 +21,26 @@ export const validateRegisterForm = args => {
...
 }
+
+export const validateEventForm = args => {
+  let Errors = {}
+  if (!args.description) Errors.description =  "this field cann't be blank"
+  if (!args.date) Errors.date =  "choose a date"
+  if (!args.time) Errors.time =  "choose time"
+  if (!args.place) Errors.place =  "choose a place"
+  
+  if (!args.description || args.description.length < 4)
```
Comment (eugenekorpan) [inline api/src/utils/validator.js]: I believe node doesn't neglect magic numbers problem :)

File: api/src/utils/validator.js [hunk 13]
```diff
@@ -21,3 +21,26 @@ export const validateRegisterForm = args => {
...
+    Errors.description =
+      "description can't be blank or shorter than 4 chars"
+  if (Object.keys(Errors).length) {
+    return {
+      success: false,
+      message: Errors
+    } 
+  }else {
```
Comment (eugenekorpan) [inline api/src/utils/validator.js]: space

File: web/src/components/Map.js [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+    super(props);
+
+    const { lat, lng } = this.props.initialCenter;
+    this.state = {
+      currentLocation: {
+        lat: lat,
+        lng: lng
+      }
+    };
+  }
```
Comment (eugenekorpan) [inline web/src/components/Map.js]: please, add blank line between functions

File: web/src/components/Map.js [hunk 15]
```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+const mapStyles = {
+  map: {
```
Comment (eugenekorpan) [inline web/src/components/Map.js]: are you sure we need to set these styles particularly here?

File: web/src/styles/create_event.css [hunk 16]
```diff
@@ -0,0 +1,287 @@
+.box {
+  position: absolute;
+}
+
+.group-title {
```
Comment (eugenekorpan) [inline web/src/styles/create_event.css]: I would strongly recommend to use nested css classes.
With global names there is a big chance someone will also somewhere create css class e.g. `title-for-form` and you'll get a conflict, mess and a lot of bugs and a lot of pain in the ass figuring out where all those styles come from

**Pull request merged**